### DISCUSSION TITLE
Appending custom tracks to albums

### DISCUSSION THREAD

**Discussion by bunnyfly** [beetbox/beets#4353] [Q&A] [answered]
I'm curious if there's a recommended way to handle custom tracks I've appended to albums that're tagged with musicbrainz?

For example, if there's a 10 song album, I might add 2 of my own custom song mixes. Before beets, I'd just set the album track count to 12 and the new songs to track 11 and 12. But since beets uses musicbrainz, the album metadata seems in conflict with what I want.

I think I could use `beet edit --all ...`, and change the tracktotals/tracks...and copy some of the other fields from the original tracks to force them together. But I don't know what impact that might have on musicbrainz integration. Would it prevent the tracks from being updated with future musicbrainz changes? Or would the next update override and erase my changes?

Maybe I should just import the original album with musicbrainz to get their tags, then scrub the musicbrainz tags and do my own edits?

Advice on a good workflow/way to handle this?!

**Comment by sampsyo** [comment, marked as answer]
Wow, interesting problem! I do think that the importer will be happiest if you start by importing only the "real" tracks and then add your own tracks after the fact. The tricky part here is that you will need to associate the new tracks with the existing album object—just changing the album name and artist name to match the existing album won't quite cut it.

Fortunately, this association is possible, but unfortunately, it's not the user-friendliest thing to do. The gist is that you will need to manipulate the `album_id` field. I would recommend doing this:

```
beet ls -af '$album_id' [name of existing album]
```

…to obtain the right id, and then do this:

```
beet modify album_id=[as above] [query describing the tracks you want to add]
```

Doing this manipulation will let you add new tracks to an existing album that you have previously tagged as a unit.

**Reply by bunnyfly** [reply]
Ah - that makes sense - I'll give that a go!

Thank you for the solution and context!! (And for the great tool in general!)
